### PR TITLE
DSL for automatically writing shell scripts

### PR DESCRIPTION
**Pull request by RomanKovalik**
Changed test providers to include start, stop functions since you can't have multiple traits overriding beforeAll and afterAll with spec2

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -111,6 +113,12 @@ publishTo in ThisBuild := {
     Some(Resolver.file("file",  new File( "maven-repo/releases" )) )
 }
 
+initialize := {
+  val required = "1.7"
```
Comment (camjo) [inline build.sbt]: Remove

File: src/main/scala/com/klout/scoozie/jobs/ShellJob.scala [hunk 2]
```diff
@@ -53,23 +60,48 @@ object ShellJob {
...
-        configuration = configBuilderImpl(configuration),
-        exec = exec,
-        argument = arguments,
-        envu45var = environmentVariables,
-        file = file,
-        archive = archive,
-        captureu45output = if (captureOutput) Some(FLAG()) else None,
-        xmlns = "uri:oozie:shell-action:0.3")
+    exec match {
+      case Left(shellString) =>
```
Comment (RomanKovalik) [inline src/main/scala/com/klout/scoozie/jobs/ShellJob.scala]: Ill rename this from shellString to oozieExecString

File: src/main/scala/com/klout/scoozie/jobs/ShellJob.scala [hunk 3]
```diff
@@ -1,13 +1,20 @@
...
-import com.klout.scoozie.dsl.{ ConfigBuilder, ConfigurationList, Job }
+import com.klout.scoozie.dsl.{ConfigBuilder, ConfigurationList, Job}
+import com.klout.scoozie.utils.WriterUtils
 import oozie.shell_0_3._
 
 import scalaxb.DataRecord
 
+case class ShellScript(script: String) extends AnyVal
+
+case class ShellScriptDescriptor(name: String, script: String)
```
Comment (camjo) [inline src/main/scala/com/klout/scoozie/jobs/ShellJob.scala]: Is this meant to be `case class ShellScriptDescriptor(name: String, script: ShellScript)`?

Comment (camjo) [inline src/main/scala/com/klout/scoozie/jobs/ShellJob.scala]: Nvm looks like its not

File: src/main/scala/com/klout/scoozie/utils/WriterUtils.scala [hunk 4]
```diff
@@ -55,10 +59,20 @@ object WriterUtils {
     assertion = path.isDefined,
     message = s"A path was not defined for the following $applicationType: $applicationName")
 
-  def createPathProperty(name: String, folderName: String) = {
-    val fileName = withXmlExtension(name)
+  def createPathProperty(name: String, folderName: String, extension: String = "xml"): (String, String) = {
```
Comment (camjo) [inline src/main/scala/com/klout/scoozie/utils/WriterUtils.scala]: ScoozieConfig.xmlExtension

File: src/test/scala/com/klout/scoozie/runner/OozieClientProvider.scala [hunk 5]
```diff
@@ -8,13 +8,14 @@ import org.apache.oozie.local.LocalOozie
 trait OozieClientProvider {
   val oozieClient: OozieClient
   def oozieCoordClient: OozieClient
+  def startOozie(): Unit
+  def stopOozie(): Unit
 }
 
 trait TestOozieClientProvider extends OozieClientProvider {
-  this: HdfsProvider =>
+  this: TestHdfsProvider =>
```
Comment (camjo) [inline src/test/scala/com/klout/scoozie/runner/OozieClientProvider.scala]: Nooooo

File: src/test/scala/com/klout/scoozie/runner/OozieClientProvider.scala [hunk 6]
```diff
@@ -8,13 +8,14 @@ import org.apache.oozie.local.LocalOozie
 trait OozieClientProvider {
   val oozieClient: OozieClient
   def oozieCoordClient: OozieClient
+  def startOozie(): Unit
```
Comment (camjo) [inline src/test/scala/com/klout/scoozie/runner/OozieClientProvider.scala]: Why are these exposed? The users of this trait shouldn't have to deal with the lifecycle issues!

File: src/test/scala/com/klout/scoozie/runner/ScoozieAppSpec.scala [hunk 7]
```diff
@@ -12,6 +12,16 @@ import org.specs2.specification.BeforeAfterAll
 import scala.concurrent.Await
 
 class ScoozieAppSpec extends Specification with BeforeAfterAll with TestOozieClientProvider with TestHdfsProvider {
+  def beforeAll() = {
+    startHdfs()
```
Comment (camjo) [inline src/test/scala/com/klout/scoozie/runner/ScoozieAppSpec.scala]: Stack your before + afters if you need to enforce ordering

Comment (camjo) [inline src/test/scala/com/klout/scoozie/runner/ScoozieAppSpec.scala]: You already require hdfs to run oozie so set up your before all and after all to reflect this ordering

File: src/main/scala/com/klout/scoozie/utils/WriterUtils.scala (hunk 4 above)
Comment (RomanKovalik) [inline src/main/scala/com/klout/scoozie/utils/WriterUtils.scala]: Yep will change

**Comment by camjo** [comment]
lgtm

File: src/test/scala/com/klout/scoozie/runner/OozieClientProvider.scala (hunk 5 above)
Comment (RomanKovalik) [inline src/test/scala/com/klout/scoozie/runner/OozieClientProvider.scala]: That was accidentally left over trying to get around not being able to use super.beforeAll

**Pull request merged**